libs/gi/db/src/thread.ts

```diff
@@ -1 +0,0 @@
-export const defThreads = navigator?.hardwareConcurrency || 4
```

Comment:
Does this change need to happen here? Max threads being the default feels the best for user perception of the site/how fast it can run

Reply:
The issue is that the `navigator` doesnt exist on server side, so having this API call in the "db" lib, which is suppose to be a pure JS lib, is technically wrong to begin with. I am pretty sure this doesn't change the functionality on the GO side, moving this API call into the UI side.

Reply:
It does change the default number of threads 
![image](https://github.com/frzyc/genshin-optimizer/assets/36019388/a40d6d85-72a6-4d11-b02b-e4b3267e756f)

Reply:
Can you add a `navigator?.hardwareConcurrency || defThreads` to wherever this is used in the regular frontend UI?